aten/src/ATen/core/dispatch/OpSchema.h

```diff
@@ -29,17 +34,40 @@ inline DeviceTypeId to_device_type_id(DeviceType device_type) {
   }
 }
 
-inline TensorParameterDispatchKey tensor_to_dispatch_key(const C10Tensor& tensor) {
+inline TensorParameterDispatchKey tensor_to_dispatch_key(const at::Tensor& tensor) {
   return TensorParameterDispatchKey{
-      to_device_type_id(tensor.impl()->device_type()),
+      to_device_type_id(tensor.getIntrusivePtr()->device_type()),
```

Comment:
Now that you're using `at::Tensor`, no need to go through the TensorImpl to get the device type, just say `tensor.device().type()`